apps/frontend/src/app/3_organisms/BorrowLoan/components/ExtendLoanForm/ExtendLoanForm.tsx

```diff
@@ -0,0 +1,394 @@
+};
+
+export const ExtendLoanForm: FC<ExtendLoanFormProps> = ({ loan }) => {
+  const { checkMaintenance, States } = useMaintenance();
+  const dappLocked = checkMaintenance(States.FULLD2);
+
+  const [nextRolloverDate, setNextRolloverDate] = useState(
+    dayjs(loan.rolloverDate * 1000)
+      .add(1, 'day')
+      .unix(),
+  );
+  const [isCalendarVisible, setIsCalendarVisible] = useState(false);
+  const [collateralAssetPrice, setCollateralAssetPrice] = useState('0');
+
+  const debtToken = useMemo(
+    () => normalizeToken(loan.debtAsset.toLowerCase()),
+    [loan.debtAsset],
+  );
+  const collateralToken = useMemo(
+    () => normalizeToken(loan.collateralAsset.toLowerCase()),
+    [loan.collateralAsset],
+  );
+  const { price: rbtcPrice } = useGetRBTCPrice();
+
+  const [useCollateral, setUseCollateral] = useState(true);
+
+  const { depositAmount, handleSubmit } = useExtendLoan(
+    loan,
+    debtToken,
+    nextRolloverDate,
+    useCollateral,
+  );
+
+  const { borrowPriceUsd, collateralPriceUsd } = useGetCollateralAssetPrice(
+    debtToken,
+    collateralToken,
+  );
+
+  const collateralChange = useMemo(() => {
+    if (useCollateral) {
+      const collateralUsdPrice =
+        collateralToken === SupportedTokens.rbtc
+          ? rbtcPrice
+          : collateralPriceUsd;
+      const loanTokenUsdPrice =
+        debtToken === SupportedTokens.rbtc ? rbtcPrice : borrowPriceUsd;
+
+      return decimalic(depositAmount)
+        .mul(loanTokenUsdPrice)
+        .div(collateralUsdPrice || 1);
+    }
+    return decimalic(0);
+  }, [
+    borrowPriceUsd,
+    collateralPriceUsd,
+    collateralToken,
+    debtToken,
+    depositAmount,
+    rbtcPrice,
+    useCollateral,
+  ]);
+
+  const newCollateralAmount = useMemo(
+    () =>
+      decimalic(loan.collateral.toString()).sub(collateralChange.toString()),
+    [collateralChange, loan.collateral],
+  );
+
+  const newTotalDebt = useMemo(
+    () =>
+      decimalic(loan.debt.toString()).add(useCollateral ? 0 : depositAmount),
+    [depositAmount, loan.debt, useCollateral],
+  );
+
+  const collateralRatio = useMemo(() => {
+    if (!nextRolloverDate) {
+      return Decimal.from(loan.collateralRatio);
+    }
+
+    const collateralUsdPrice =
+      collateralToken === SupportedTokens.rbtc ? rbtcPrice : collateralPriceUsd;
+    const totalDebtUsd = newTotalDebt.mul(
+      debtToken === SupportedTokens.rbtc ? rbtcPrice : borrowPriceUsd,
+    );
+
+    return newCollateralAmount
+      .mul(collateralUsdPrice)
+      .div(totalDebtUsd)
+      .mul(100);
+  }, [
+    borrowPriceUsd,
+    collateralPriceUsd,
+    collateralToken,
+    debtToken,
+    loan.collateralRatio,
+    newCollateralAmount,
+    newTotalDebt,
+    nextRolloverDate,
+    rbtcPrice,
+  ]);
+
+  const isValidCollateralRatio = useMemo(() => {
+    return (
+      Number(collateralAssetPrice) === 0 ||
+      collateralRatio.gte(
+        MINIMUM_COLLATERAL_RATIO_BORROWING_MAINTENANCE.mul(100),
```

Comment:
Are you sure about using `MINIMUM_COLLATERAL_RATIO_BORROWING_MAINTENANCE`? I think it should be used only for the calculation of `liquidationPrice` and you should use `initial collateral ratio` (150%/300%) instead (look at `minimumCollateralRatio` in `AdjustLoanForm`).